pkg/controller/utils/cache/http.go

```diff
@@ -40,26 +42,44 @@ func GetHTTPClient(accessToken []byte, transport *http.Transport) (*http.Client,
 	return http.DefaultClient, nil
 }
 
-// Populates a TLS config struct based specified input.
-func GetTLSCConfig(c client.Client, skipCertVerify bool) (*tls.Config, error) {
+// Log mutex
+var logRouterCertError sync.Once
+
+// Log that there was a problem obtaining the default ingress router CA
+// certificate.  Only log once as the same error is likely to happen
+// over and over again.
+func logIngressRouterCertError(logger logr.Logger, err error) {
```

Comment:
Tim, i am concerned that printing it once may get lost and it would become hard to debug. If possible, could we leave a breadcrumb of information in the error message when we detect that request failed? That is by labeling/printing the client/transport struct in the error message.

Reply:
@mezarin I just want to be sure I understand what you are asking for.... you want to print the golang struct for the client and transport in the error message that we print to the pod log?

I wasn't sure if you were referring to the pod log or the CR log.

Reply:
Hi Tim, It would be the kabanero stack controller pod log. If there is a failure with the http request that used transport.TLSClientConfig that is nil or not ... the error we print when the request failed should include it so we know if it was set or not.

Reply:
Hi Ed - it was pretty easy to make this change to the HTTP path.  I did not make the change to the GitHub path because the TLSClientConfig check is a couple function calls beneath where the client is used.  We can make that change later if we decide it's necessary - maybe we can wrap the GitClient or something so that we can retrieve the TLS config later.